### PR TITLE
Show external data sources in codebook

### PR DESCRIPTION
**Pull request by wwqrd**
Primarily this adds external data to the Codebook. User is shown a list of variable names.

I also took the time to refactor the external data HOC and external data verification functions as there was unnecessary duplication.

Resolves: https://github.com/complexdatacollective/Architect/issues/649

### CODE REVIEW

**Comment by wwqrd** [comment]
@rebeccamadsen Please hold on the review, I think I may have found a bug already

**Comment by wwqrd** [comment]
@rebeccamadsen false alarm, all good (afaik!)

**Review by rebeccamadsen — APPROVED** [review]
This looks great to me! I noticed two spelling errors, but I don't think they are related to your changes, so I can create an issue for those if you'd prefer.

"attmpted" when uploading an invalid csv file:
spelling

"accurary" when viewing the roster name generator:
spelling2

Everything I've explored with the external data sources seems to be awesome though.

File: src/ducks/modules/protocol/utils/dialogs.js [hunk 1]
```diff
@@ -10,18 +11,23 @@ export const invalidAssetErrorDialog = (e, filePath) => {
...
       </p>
       <p>
-        <strong>Reminder:</strong> Network Canvas supports importing network data in
-        either JSON or CSV format. JSON files must contain a &quot;nodes&quot; and/or
-        &quot;edges&quot; list in the root object, and CSV data should contain a header
-        row comprising variable names, and be in the format of a node or edge list.
-      </p>
-      <p>
-        Variable names must use only letters, numbers, the symbols ._-: and contain no spaces.
+        The file you attempted to import is not in a format supported by
```
Comment (rebeccamadsen) [inline src/ducks/modules/protocol/utils/dialogs.js:14]: This ends up repeating the same sentence twice, right? "The file you attempted to import is not in a format supported by Network Canvas." Is that intentional?

Comment (wwqrd) [reply src/ducks/modules/protocol/utils/dialogs.js:14]: :facepalm:

Comment (wwqrd) [reply src/ducks/modules/protocol/utils/dialogs.js:10]: Have removed the duplicate line.

**Review by rebeccamadsen — APPROVED** [review]
Looks great to me!

**Pull request merged**